README.md

```diff
@@ -123,24 +123,79 @@ Alternatively, you can install it in your entire system instead (not recommended
 gem install arkana
 ```
 
-# Usage
+# Basic Usage
 
-Arkana requires the declaration of a YAML config file. Although you can name it whatever, the convention is to name it `.arkana.yml`. See [template.yml](/template.yml) for practical examples.
+Arkana requires the declaration of a YAML config file. Although you can name it whatever, the convention is to name it `.arkana.yml`. See [template.yml](/template.yml) for complete options.
 
-Once you have create your config file, you can run Arkana:
+Once you have created your config file, you can run Arkana:
 
 ```sh
 Usage: arkana [options]
     -c /path/to/your/.arkana.yml,    Path to your config file. Defaults to '.arkana.yml'
-        --config-filepath
     -e /path/to/your/.env,           Path to your dotenv file. Defaults to '.env' if one exists.
-        --dotenv-filepath
-    -f, --flavor FrostedFlakes       Flavors are useful, for instance, when generating secrets for white-label projects. See the README for more information
-    -i debug,release,                Optionally pass the environments that you want Arkana to generate secrets for. Useful if you only want to build a certain environment, e.g. just Debug in local machines, while only building Staging and Release in CI. Separate the keys using a comma, without spaces. When omitted, Arkana generate secrets for all environments.
-        --include-environments
-    -l, --lang kotlin                Language to produce keys for, e.g. kotlin, swift. Defaults to 'swift'. See the README for more information
+    -l kotlin                        Language to produce keys for, e.g. kotlin, swift. Defaults to 'swift'.
 ```
 
+
+> [!NOTE]
+> For complete set of args look at the [options](#options) section.
+
+
+### Config File
+
+The `arkana.yml` would typically contain 3 important sections
+ - **Environments** : This is typically where you specify `debug`, `release` or other environments which you wish to create 
+ - **Environment Secrets**: This is where you declare the keys which will be ultimately exposed to your app like `apiKey`
+ - **Global Secrets** : Here you'd declare keys which are the same across all environments.
+
+### Environment File
+
+The environment(.env) file contains the actual secrets for each environment. While config file declares the keys, they are assigned encrypted values from this file.
```

Comment:
These actually aren't required, and it's advised that no one commits them to git 😅 I'll add a note about this!